packages/functions/src/aggregator/handle-buffer-queue-message.js

```diff
@@ -0,0 +1,118 @@
+import * as Sentry from '@sentry/serverless'
+
+import { createClient as createBufferStoreClient } from '@w3filecoin/core/src/store/aggregator-buffer-store.js'
+import { createClient as createBufferQueueClient, decodeMessage } from '@w3filecoin/core/src/queue/buffer-queue.js'
+import { createClient as createAggregateOfferQueueClient } from '@w3filecoin/core/src/queue/aggregate-offer-queue.js'
+import * as aggregatorEvents from '@web3-storage/filecoin-api/aggregator/events'
+
+import { mustGetEnv } from '../utils.js'
+
+Sentry.AWSLambda.init({
+  environment: process.env.SST_STAGE,
+  dsn: process.env.SENTRY_DSN,
+  tracesSampleRate: 1.0,
+})
+
+/**
+ * On buffer queue messages, reduce received buffer records into a bigger buffer.
+ * - If new buffer does not have enough load to build an aggregate, it is stored
+ * and requeued for buffer reducing
+ * - If new buffer has enough load to build an aggregate, it is stored and queued
+ * into aggregateOfferQueue. Remaining of the new buffer (in case buffer bigger
+ * than maximum aggregate size) is re-queued into the buffer queue.
+ *
+ * @param {import('aws-lambda').SQSEvent} sqsEvent
+ */
+async function handleBufferQueueMessage (sqsEvent) {
+  // if one we should put back in queue
+  if (sqsEvent.Records.length === 1) {
+    return {
+      batchItemFailures: sqsEvent.Records.map(r => ({
```

Comment:
What is the difference between this and just throwing and error?

Reply:
so that we can return `200` and account for metrics in a different way if we would like to. This is because it is possible to receive 1 item in the batch if not enough trhoughput